# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/debugger/Frame-inl.h
# Commit: 2d470b425ce9
# Full Hash: 2d470b425ce9d8fc12c89f087d76ff1a337619ab
# Author: Logan Smyth <loganfsmyth@gmail.com>
# Date: 2020-06-15 21:48:38
# Description:
#   Bug 1644699 - Allow Debugger.Frame to have GeneratorInfo that is closed. r=arai
#   
#   The GeneratorInfo data associated with a Debugger.Frame is not guaranteed
#   to be cleared when a generator transitions from running/suspended to closed.
#   In the current codebase, we have broadly assumed that to be true, which was
# ==============================================================================

diff -r d56e40dc9c0f -r 2d470b425ce9 js/src/debugger/Frame-inl.h
--- a/js/src/debugger/Frame-inl.h	Mon Jun 15 14:33:28 2020 +0000
+++ b/js/src/debugger/Frame-inl.h	Mon Jun 15 14:26:35 2020 +0000
@@ -13,13 +13,13 @@
 
 #include "NamespaceImports.h"  // for Value
 
-inline bool js::DebuggerFrame::hasGenerator() const {
+inline bool js::DebuggerFrame::hasGeneratorInfo() const {
   return !getReservedSlot(GENERATOR_INFO_SLOT).isUndefined();
 }
 
 inline js::DebuggerFrame::GeneratorInfo* js::DebuggerFrame::generatorInfo()
     const {
-  MOZ_ASSERT(hasGenerator());
+  MOZ_ASSERT(hasGeneratorInfo());
   return static_cast<GeneratorInfo*>(
       getReservedSlot(GENERATOR_INFO_SLOT).toPrivate());
 }